src/components/LoginComponent/LoginComponent.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useQuery_UserLazyQuery } from '../../api/graphql/__generated__/graphql-types';
+import { useLoginContext } from '../../context/LoginContext/LoginContext';
+import { Types } from '../../context/LoginContext/loginReducer';
+import { emailValidation } from '../../helpers/validators';
+import Button from '../Button/Button';
+import ErrorMessage from '../ErrorMessage/ErrorMessage';
+import HeadingTitle from '../Text/HeadingTitle/HeadingTitle';
+import { Container } from './LoginComponentStyled';
+import TextField from '../TextField/TextField';
+
+interface LoginComponentProps {
+  title: string;
+}
+
+const LoginComponent = ({ title }: LoginComponentProps) => {
+  const [isButtonDisabled, setIsButtonDisabled] = useState<boolean>(true);
+  const [email, setEmail] = useState<string>('');
+  const [isError, setIsError] = useState(false);
+
+  const navigate = useNavigate();
+  const { dispatch } = useLoginContext();
+
+  const [getUserLazyResults, { data, error, loading }] =
+    useQuery_UserLazyQuery();
+
+  useEffect(() => {
+    if (data) {
+      if (data!.users.length > 0) {
+        dispatch({
+          type: Types.SET_USER,
+          payload: { userData: data!.users[0] },
+        });
+        navigate('/');
+        setIsError(false);
+      } else {
+        setIsError(true);
+        setIsButtonDisabled(false);
+      }
+    }
+  }, [data, loading]);
+
+  const handleClick = () => {
+    setIsButtonDisabled(true);

```

Comment:
Could you tie this disabled into the loading state of the lazy query?

Reply:
Yeah sure, but here i have a disabled button when is not a valid email but i removed it and now i handle this like so: 
```
useEffect(() => {
    const isValidEmail = emailValidation(email);
    if (data) {
      if (isUser && isValidEmail) {
        dispatch({
          type: Types.LOGIN_USER,
          payload: { userData: data.users[0] },
        });
        navigate('/');
        setIsError(false);
      } else {
        setIsError(true);
      }
    }
  }, [data, loading]);
```
and my button is:
```
<Button
        title={loading ? 'Loading...' : 'Login'}
        onClick={handleClick}
        disabled={loading}
      />
```